content/spin/kubernetes.md

```diff
@@ -0,0 +1,382 @@
+```
+
+Next update to the latest version:
+
+```bash
+az extension update --name aks-preview
+```
+
+Register the WasmNodePoolPreview feature
+
+```bash
+az feature register --namespace "Microsoft.ContainerService" --name "WasmNodePoolPreview"
+```
+
+This will take a few minutes to complete. You can verify it’s done when this command returns *Registered*
+
+```bash
+az feature show --namespace "Microsoft.ContainerService" --name "WasmNodePoolPreview"
+```
+
+Finally refresh the registration of the ContainerService
+
+```bash
+az provider register --namespace Microsoft.ContainerService
+```
+
+Once the service is registered, the next step is to add a Wasm/WASI nodepool to an existing AKS cluster. If a cluster doesn’t already exist, follow Azure’s [documentation](https://learn.microsoft.com/en-us/azure/aks/learn/quick-kubernetes-deploy-portal?tabs=azure-cli) to create a new cluster.
+
+```bash
+az aks nodepool add \
+    --resource-group myResourceGroup \
+    --cluster-name myAKSCluster \
+    --name mywasipool \
+    --node-count 1 \
+    --workload-runtime WasmWasi
+```
+
+You can verify the workloadRuntime using the following command
+
+```bash
+az aks nodepool show -g myResourceGroup --cluster-name myAKSCluster -n mywasipool --query workloadRuntime
+```
+
+The next set of commands uses kubectl to create the necessary runtimeClass. If you don’t already have kubectl configured with the appropriate credentials, you can retrieve them with this command
+
+```bash
+az aks get-credentials -n myakscluster -g myresourcegroup
+```
+
+Find the name of the nodepool
+
+```bash
+kubectl get nodes -o wide
+```
+
+Then retrieve detailed information on the appropriate nodepool and verify among it’s labels is “kubernetes.azure.com/wasmtime-spin-v1=true”
+
+```bash
+kubectl describe node aks-mywasipool-12456878-vmss000000
+```
+
+Create a file wasm-runtimeclass.yml and populate with the following information
+
+```yaml
+apiVersion: node.k8s.io/v1
+kind: RuntimeClass
+metadata:
+  name: "wasmtime-spin"
+handler: "spin"
+scheduling:
+  nodeSelector:
+    "kubernetes.azure.com/wasmtime-spin": "true"
+```
+
+Then register the runtime class with the cluster
+
+```yaml
+kubectl apply -f wasm-runtimeclass.yaml
+```
+
+{{ blockEnd }}
+
+{{ startTab "K3d"}}
+
+## Setup K8s for Spin
+
+### Introduction
+
+[K3d](https://k3d.io/) is a lightweight Kubernetes installation.
+
+### Known Limitations
+
+- Each Pod will be constantly running it’s own HTTP listener which adds overhead vs Fermyon Cloud.
+- You can run containers and wasm modules on the same node, but you can't run containers and wasm modules on the same pod.
+
+### Setup
+
+Ensure both Docker and k3d are installed. Then [enable Containerd](https://docs.docker.com/desktop/containerd/) for Docker in Settings → Experimental → Use containerd for pulling and storing images.
+
+![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/0b572381-09cc-4baf-b210-88c6e88cbc5f/Untitled.png)
```

Comment:
This image is not rendering in the preview
![Screenshot 2023-04-14 at 12 22 16 pm](https://user-images.githubusercontent.com/9831342/231925554-3a163702-5d43-4e80-9a3d-d5767fafe075.png)

![Screenshot 2023-04-14 at 12 23 17 pm](https://user-images.githubusercontent.com/9831342/231925705-ad475df2-aab4-47e7-a691-0600db454995.png)

If you prefer, please upload any images to this website's image folder at < https://github.com/fermyon/developer/tree/main/static/image > and then link to the images in the markdown i.e.

```md
![my image description](/static/image/my_image.png)
```

Reply:
I decided to nuke all the images. Missed this one